modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,260 @@
+= Reusing Types from a JSON Schema
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+JSON schema is a standard providing a format for what JSON data is required for a given application and how to interact with it. Many existing data formats or types are expressed using this standard. In particular, REST APIs described by a RAML file are likely to have their types defined in JSON schemas.
+
+To reuse predefined types, you can load the JSON schema into DataWeave to make the types structure available for use in your scripts.
+
+The JSON schema loader parses schemas following the https://json-schema.org/specification-links.html#draft-7[JSON Schema Draft 7ˆ].
+
+== Import a Type from a JSON Schema
+
+The following example shows a JSON schema (`Person.json`) in the resources directory: 
+
+.JSON Schema (`example/schema/Person.json`):
+[source,JSON]
+----
+{
+  "$id": "https://example.com/person.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Person",
+  "type": "object",
+  "properties": {
+    "firstName": {
+      "type": "string",
+      "description": "The person's first name."
+    },
+    "lastName": {
+      "type": "string",
+      "description": "The person's last name."
+    },
+    "age": {
+      "description": "Age in years which must be equal to or greater than zero.",
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "required": ["firstName", "lastName"]
+}
+----
+
+The following example shows how you can use the JSON schema loader to import the type:
+
+.DataWeave Script Header:
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Import Syntax
+
+To import JSON schema types, use the following syntax, where:
+
+* `typeToImport`: You can use `*` alone to import all types defined in the schema or you can import a single type from the schema using, for example, `Root`. You can also import JSON schema types with a different name, for example, `Root as Person`, which enables you to reference the type with that name in the script.
+* `pathToJsonSchema`: To specify the path to the schema file, replace the file separators with `::` and remove the `.json` extension from the file name. For example, if the path to the schema is `example/schema/Person.json`, use `example::schema::Person`.
+
+----
+import _typesToImport_ from jsonschema!_pathToJsonSchema_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+== Use Your Types in a DataWeave Script
+
+Having the import directive from the above example in the script header loads all the existing types in the JSON schema. In `import * from jsonschema!example::schema::Person`, the only existing type is the `Root` type, specified at the root which describes an Object that can have three properties (`firstName`, `lastName`, and `age`). It is equivalent to declaring the following type in your DataWeave script:
```

Comment:
Having -> Including?

Avoid directional terminology (above).

The second sentence is a long drink of water. Can you split it in two?

Avoid "it is" construction.

Reply:
Rewording.